_posts/2018-05-02-load-balanced-smart-proxies-with-haproxy.md

```diff
@@ -0,0 +1,151 @@
+
+| Service    | Port | Mode | Balance mode | Destination |
+| ---------- | ---- | ---- | ------------ | ----------- |
+| HTTP       | 80   | TCP  | `roundrobin` | port 80 on all proxies |
+| Anaconda   | 8000 | TCP  | `roundrobin` | port 8000 on all proxies|
+| HTTPS      | 443  | TCP  | `source`     | port 443 on all proxies |
+| RHSM       | 8443 | TCP  | `roundrobin` | port 8443 on all proxies |
+| AMQP       | 5647 | TCP  | `roundrobin` | port 5647 on all proxies |
+| Puppet     | 8140 | TCP  | `roundrobin` | port 8140 on all proxies |
+| PuppetCA   | 8141 | TCP  | `roundrobin` | port 8140 on the first proxy |
+| SmartProxy | 9090 | TCP  | `roundrobin` | port 9090 on all proxies |
+| Docker     | 5000 | TCP  | `roundrobin` | port 5000 on all proxies |
+
+[You can find an example `haproxy.cfg` in the `forklift` repository.](https://github.com/theforeman/forklift/blob/master/roles/haproxy/templates/haproxy.cfg.j2)
+
+## Configuring Clients
+
+As far as clients are concerned, they should only know about `smartproxy.example.net`, and should never need to connect directly to a smartproxy. That has to be configured for Puppet, goferd, and for RHSM/Yum.
+
+### Using katello-client-bootstrap
+
+[katello-client-bootstrap](https://github.com/Katello/katello-client-bootstrap) is designed to configure a machine to be used with an exiting Katello environment. It can also configure Puppet properly for us and supports [setting a separate Puppet CA and Puppet CA port in current Git](https://github.com/Katello/katello-client-bootstrap/pull/250).
+
+To attach clients to the load-balanced proxy, we just have to pass `--server smartproxy.example.net` and `--puppet-ca-port 8141`, `bootstrap.py` will take care of the rest.
+
+### Manual registration
+
+Instead of using `bootstrap.py`, the client can also be registered manually. However, we recommend using `bootstrap.py` since it handles these extra manual steps for you.
+
+After installing `katello-ca-consumer` RPM from `http://smartproxy.example.net/pub/katello-ca-consumer-latest.noarch.rpm` `/etc/rhsm/rhsm.conf` and `/etc/yum.repos.d/redhat.repo` will contain the hostname of whichever proxy served the initial request and not `smartproxy.example.net` as we'd like it. Thus, when registering the system with `subscription-manager`, we have to pass `--serverurl=https://smartproxy.example.net:8443/rhsm --baseurl=https://smartproxy.example.net/pulp/repos` to update those two config files.
+
+Since PuppetCA is reachable on a dedicated port, all Puppet agents need to be configured to talk to `smartproxy.example.net` on port 8140 for Puppet requests and port 8141 for PuppetCA requests.
+This can be achieved by modifying `puppet.conf` to include the following lines:
+
+```
+server = smartproxy.example.net
+ca_server = smartproxy.example.net
```

Comment:
Technically I think this line is redundant but doesn't hurt either.

Reply:
Yeah, but I kinda preferred being explicit here.